packages/brick_supabase/lib/src/query_supabase_transformer.dart

```diff
@@ -188,7 +188,11 @@ class QuerySupabaseTransformer<_Model extends SupabaseModel> {
 
     return [
       {
-        queryKey: '${_compareToSearchParam(condition.compare)}.${condition.value}',
+        queryKey: condition.compare == Compare.inList
+            ? (condition.value is Iterable && (condition.value as Iterable).isNotEmpty
+                ? 'in.(${(condition.value as Iterable).map(_quoteSupabaseValue).join(',')})'
+                : 'in.()')
+            : '${_compareToSearchParam(condition.compare)}.${condition.value}',
```

Comment:
This, combined with your later change, makes the `_compareToSearchParam`  switch exhaustive but unused. Instead, consider generating the value separately

```dart
final value = condition.value is Iterable ? '(${condition.value.map((v) => v.toString()).join(',')})' : condition.value;
```

```suggestion
        queryKey: '${_compareToSearchParam(condition.compare)}.$value',
```